lib/shopify_api/graphql/bulk_fetch.ex

```diff
@@ -0,0 +1,82 @@
+defmodule ShopifyAPI.GraphQL.BulkFetch do
+  alias ShopifyAPI.AuthToken
+
+  @spec process(AuthToken.t(), String.t(), integer()) :: any()
+  def process(%AuthToken{} = token, query, polling_rate \\ 100) do
+    with :ok <- create(token, query),
+         {:ok, url} <- poll_till_completed(token, polling_rate),
+         {:ok, %{body: jsonl}} <- HTTPoison.get(url) do
+      {:ok, parse_bulk_response(jsonl)}
+    else
+      error -> error
+    end
+  end
+
+  defp create(token, query) do
+    query_string = Jason.encode!(query)
```

Comment:
🔥 we should use the json serializer config

here: https://github.com/pixelunion/elixir-shopifyapi/blob/master/lib/shopify_api/json_serializer.ex#L15